tests/garage/experiment/test_snapshotter.py

```diff
@@ -19,41 +24,78 @@
+
+        algo = VPG(
+            env_spec=env.spec,
+            policy=policy,
+            baseline=baseline,
+            max_path_length=100,
+            discount=0.99,
+            optimizer_args=dict(tf_optimizer_args=dict(learning_rate=0.01, )))
+
+        runner.setup(algo, env)
+        runner.train(n_epochs=5, batch_size=10)
+
+
+class TestSnapshotter(TfGraphTestCase):
     def setup_method(self):
-        self.snapshot_dir = tempfile.TemporaryDirectory()
-        self.snapshotter = Snapshotter()
+        super().setup_method()
+        self.temp_dir = tempfile.TemporaryDirectory()
 
     def teardown_method(self):
-        self.snapshotter.reset()
-        self.snapshot_dir.cleanup()
-
-    def test_set_snapshot_dir(self):
-        self.snapshotter.snapshot_dir = self.snapshot_dir.name
-        assert self.snapshotter.snapshot_dir == self.snapshot_dir.name
+        super().teardown_method()
+        self.temp_dir.cleanup()
 
     @pytest.mark.parametrize('mode, files', [*configurations])
     def test_snapshotter(self, mode, files):
-        self.snapshotter.snapshot_dir = self.snapshot_dir.name
+        dir = self.temp_dir
+        snapshotter = Snapshotter(dir.name, mode, 2)
 
-        self.snapshotter.snapshot_mode = mode
-        assert self.snapshotter.snapshot_mode == mode
-        self.snapshotter.snapshot_gap = 2
-        assert self.snapshotter.snapshot_gap == 2
+        assert snapshotter.snapshot_dir == dir.name
+        assert snapshotter.snapshot_mode == mode
+        assert snapshotter.snapshot_gap == 2
 
         snapshot_data = [{'testparam': 1}, {'testparam': 4}]
-        self.snapshotter.save_itr_params(1, snapshot_data[0])
-        self.snapshotter.save_itr_params(2, snapshot_data[1])
+        snapshotter.save_itr_params(1, snapshot_data[0])
+        snapshotter.save_itr_params(2, snapshot_data[1])
 
         for f, num in files.items():
-            filename = osp.join(self.snapshot_dir.name, f)
+            filename = osp.join(dir.name, f)
             assert osp.exists(filename)
             with open(filename, 'rb') as pkl_file:
                 data = pickle.load(pkl_file)
                 assert data == snapshot_data[num]
 
     def test_invalid_snapshot_mode(self):
         with pytest.raises(ValueError):
-            self.snapshotter.snapshot_dir = self.snapshot_dir.name
-            self.snapshotter.snapshot_mode = 'invalid'
-            self.snapshotter.save_itr_params(2, {'testparam': 'invalid'})
+            snapshotter = Snapshotter(
+                snapshot_dir=self.temp_dir.name, snapshot_mode='invalid')
+            snapshotter.save_itr_params(2, {'testparam': 'invalid'})
+
+    def test_load(self):
+        snapshot_config = SnapshotConfig(
+            snapshot_dir=self.temp_dir.name,
+            snapshot_mode='all',
+            snapshot_gap=1)
+        sample_task(snapshot_config)
+        snapshotter = Snapshotter()
+
+        super().teardown_method()
```

Comment:
please make these different test cases, rather than using teardown/setup manually. that is an abuse of the test framework.